staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation_test.go

```diff
@@ -3014,6 +3020,314 @@ func TestValidateFieldPath(t *testing.T) {
+			schema: mustSchema(`
+                type: array
+                x-kubernetes-list-type: atomic
+                items:
+                    type: object
+                    properties:
+                        bar:
+                            type: string
+                            x-kubernetes-validations:
+                            - rule: self == "baz"
+                              message: "gotta be baz"
+                `),
+			// CEL error comes from uncorrelatable portion of the schema,
+			// but it should be ratcheted anyway because it is the descendent
+			// or an unchanged correlatable node
+			oldObj: mustUnstructured(`
+                - bar: bar
+            `),
+			newObj: mustUnstructured(`
+                - bar: bar
+            `),
+			warnings: []string{
+				`root[0].bar: Invalid value: "string": gotta be baz`,
+			},
+		},
+		{
+			name: "ratchet error triggered by normal CEL expression from an uncorrelatable part of the schema if one of its sublings has changed",
+			schema: mustSchema(`
+                type: object
+                properties:
+                    stringField:
+                        type: string
+                    atomicArray:
+                        type: array
+                        x-kubernetes-list-type: atomic
+                        items:
+                            type: object
+                            properties:
+                                bar:
+                                    type: string
+                                    x-kubernetes-validations:
+                                    - rule: self == "baz"
+                                      message: "gotta be baz"
+                `),
+			oldObj: mustUnstructured(`
+                stringField: foo
+                atomicArray:
+                - bar: bar
+            `),
+			newObj: mustUnstructured(`
+                stringField: changed but ratcheted
+                atomicArray:
+                - bar: bar
+            `),
+			warnings: []string{
+				`root.atomicArray[0].bar: Invalid value: "string": gotta be baz`,
+			},
+		},
+		{
+			name: "we can't ratchet an error triggered by a normal CEL expression from an uncorrelatable part of the schema if one of its parent nodes has changed",
+			schema: mustSchema(`
+                type: array
+                x-kubernetes-list-type: atomic
+                items:
+                    type: object
+                    properties:
+                        bar:
+                            type: string
+                            x-kubernetes-validations:
+                            - rule: self == "baz"
+                              message: "gotta be baz"
+            `),
+			// CEL error comes from uncorrelatable portion of the schema,
+			// but it should be ratcheted anyway because it is the descendent
+			// or an unchanged correlatable node
```

Comment:
This comment seems not for this test case?

Reply:
Nope, it is on the correct case. The CEL error is on the schema within the items specifically, so it is not correlatable. But it is ratcheted anyway because it has a correlatable parent which is completely unchanged.